engine/src/main/java/com/agiletec/aps/system/common/tree/TreeNode.java

```diff
@@ -18,197 +18,223 @@
-		this._children = newChildren;
+		this._childrenCodes = newChildren;
 	}
-	
+
 	@Override
 	public int getPosition() {
 		return _position;
 	}
+
 	protected void setPosition(int position) {
 		this._position = position;
 	}
-	
+
 	@Override
 	public ApsProperties getTitles() {
 		return _titles;
 	}
-	
+
 	/**
-	 * Set the titles of the node. 
-	 * @param titles A set of properties with the titles, 
-	 * where the keys are the codes of language.
+	 * Set the titles of the node.
+	 *
+	 * @param titles A set of properties with the titles, where the keys are the
+	 * codes of language.
 	 */
 	public void setTitles(ApsProperties titles) {
 		this._titles = titles;
 	}
-	
+
 	@Override
 	public void setTitle(String langCode, String title) {
 		this.getTitles().setProperty(langCode, title);
 	}
-	
+
 	@Override
 	public String getTitle(String langCode) {
 		return this.getTitles().getProperty(langCode);
 	}
-	
+
 	@Override
 	public String getFullTitle(String langCode) {
 		return this.getFullTitle(langCode, " / ");
 	}
-	
+
 	@Override
 	public String getFullTitle(String langCode, String separator) {
 		return this.getFullTitle(langCode, separator, false);
 	}
-	
+
 	@Override
 	public String getShortFullTitle(String langCode) {
 		return this.getShortFullTitle(langCode, " / ");
 	}
-	
+
 	@Override
 	public String getShortFullTitle(String langCode, String separator) {
 		return this.getFullTitle(langCode, separator, true);
 	}
-	
+
 	protected String getFullTitle(String langCode, String separator, boolean shortTitle) {
 		String title = this.getTitles().getProperty(langCode);
-		if (null == title) title = this.getCode();
-		if (this.isRoot()) return title;
+		if (null == title) {
+			title = this.getCode();
+		}
+		if (this.isRoot()) {
+			return title;
+		}
 		ITreeNode parent = this.getParent();
 		while (parent != null && parent.getParent() != null) {
 			String parentTitle = "..";
 			if (!shortTitle) {
 				parentTitle = parent.getTitles().getProperty(langCode);
-				if (null == parentTitle) parentTitle = parent.getCode();
+				if (null == parentTitle) {
+					parentTitle = parent.getCode();
+				}
 			}
 			title = parentTitle + separator + title;
-			if (parent.isRoot()) return title;
+			if (parent.isRoot()) {
+				return title;
+			}
 			parent = parent.getParent();
 		}
 		return title;
 	}
-	
+
 	@Override
 	public String getPath() {
 		return this.getPath("/", true);
 	}
-	
+
 	@Override
 	public String getPath(String separator, boolean addRoot) {
 		String[] pathArray = this.getPathArray(addRoot);
 		StringBuilder path = new StringBuilder();
 		for (int i = 0; i < pathArray.length; i++) {
 			String code = pathArray[i];
-			if (i>0) path.append(separator);
+			if (i > 0) {
+				path.append(separator);
+			}
 			path.append(code);
 		}
 		return path.toString();
 	}
-	
+
 	@Override
 	public String[] getPathArray() {
 		return this.getPathArray(true);
 	}
-	
+
 	@Override
 	public String[] getPathArray(boolean addRoot) {

```

Comment:
Short comment on the goal of this method.

Reply:
Ok, I'll write a java comment